components/Common/MetaBar/index.tsx

```diff
@@ -0,0 +1,116 @@
+import { CodeBracketIcon } from '@heroicons/react/24/outline';
+import { type Heading } from '@vcarl/remark-headings';
+import Image from 'next/image';
+import {
+  useMemo,
+  type ComponentProps,
+  type FC,
+  type PropsWithChildren,
+} from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import AvatarGroup from '@/components/Common/AvatarGroup';
+import LocalizedLink from '@/components/LocalizedLink';
+
+import styles from './index.module.css';
+
+type MetaBarProps = {
+  date: Date;
+  readingTime: string;
```

Comment:
So, afaik most of these items should be optional. Such as reading time, viewsASURL, sourceURL, addedInVersion, etc...

The Metadata Tab is a generic Tab, hence I believe we should design it in a generic way that should be able to be used by Layouts as they wish.

I would actually suggest to redesign the MetadataProps to support something like:

```tsx
type MetadataItem {
  label: string;
  content: React.ReactNode;
}

type MetadataBarProps = {
  items: MetadataItem[];
  headings?: {
     items: Heading[];
     depth?: number; // have a default depth
  }
}
```

I'd say the only always required item is the "headings" (as they're used on blog posts, learn pages, regular pages, etc)